packages/table/src/cell/formats/truncatedFormat.tsx

```diff
@@ -202,6 +219,32 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
                 return detectTruncation
                     ? this.state.isTruncated
                     : truncateLength > 0 && content.length > truncateLength;
+            case TruncatedPopoverMode.WHEN_TRUNCATED_APPROX:
+                if (!detectTruncation) {
+                    return truncateLength > 0 && content.length > truncateLength;
+                }
+                const {
+                    getApproximateCharWidth: approxCharWidth,
+                    getApproximateLineHeight: approxLineHeight,
+                    getCellHorizontalPadding: horizontalPadding,
+                    getNumBufferLines: numBufferLines,
+                } = TruncatedFormat.resizeRowsByApproximateHeightDefaults;
+
+                const cellWidth = parseInt(this.props.parentCellWidth, 10);
+                const approxCellHeight = Utils.getApproxCellHeight(
+                    content,
+                    cellWidth,
+                    approxCharWidth,
+                    approxLineHeight,
+                    horizontalPadding,
+                    numBufferLines,
+                );
+
+                let shouldTruncate = false;
+                if (approxCellHeight > parseInt(this.props.parentCellHeight, 10)) {
```

Comment:
Same.

Should probably just return early on L223 if either of these doesn't exist.